### DISCUSSION TITLE
Detection of Selenium Mouse and Keyboard Inputs - App Opens Multiple Browser Windows for Simultaneous Scrapes.

### DISCUSSION THREAD

**Discussion by G-markier** [seleniumbase/SeleniumBase#2302] [Q&A] [answered]
Hi
We have an app that opens multiple windows to scrape simultaneously using different proxies for each window. Sometimes we get 404 errors for pages that we request even though we know that the pages exists. Based on readings here I am suspecting that that our mouse and keyboard entries (which is using Selenium) is being detected.

Does Seleniumbase provide an alternative for mouse and keyword inputs without using those provided by base Selenium? If not, are there any good alternative for an app like ours with multiple browser windows?

Thanks

**Comment by mdmintz** [comment, marked as answer]
Regular manual clicking or selenium clicking during UC Mode can get you detected because those websites aren't looking specifically at the click, but they are looking at your environment settings / Chrome variables / etc. The `driver.uc_click(selector)` method disconnects chromedriver from Chrome immediately after the click for a very short time... If a website looks for traces of chromedriver, then it won't find it if chromedriver disconnected fast enough.

However, if you want to do some manual actions during UC Mode, first call `driver.reconnect(timeout)`, which will disconnect chromedriver from Chrome for that many seconds, and then you can perform manual actions safely before your time runs out.

If you're using SeleniumBase to set multiple `proxies for each window` (with authentication), make sure you have the `multi_proxy` / `--multi-proxy` option set. Example:

```python
from parameterized import parameterized
from seleniumbase import BaseCase
BaseCase.main(__name__, __file__, "-n3")

class ProxyTests(BaseCase):
    @parameterized.expand(
        [
            ["user1:pass1@host1:port1"],
            ["user2:pass2@host2:port2"],
            ["user3:pass3@host3:port3"],
        ]
    )
    def test_multiple_proxies(self, proxy_string):
        self.get_new_driver(
            undetectable=True, proxy=proxy_string, multi_proxy=True
        )
        self.driver.get("https://browserleaks.com/webrtc")
        self.sleep(30)
```
